libs/community/langchain_community/document_loaders/sqlalchemy.py

```diff
@@ -0,0 +1,144 @@
+import functools
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class SQLAlchemyLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        url: str,
+        page_content: Optional[Callable] = None,
+        metadata: Optional[Callable] = None,
```

Comment:
See comment below
```suggestion
        page_content: Optional[Callable] = self.page_content_default,
        metadata: Optional[Callable] = self.metadata_default,
```

Reply:
Both functions are now used as default values, implementation is slightly different.

https://github.com/langchain-ai/langchain/blob/df35ef7e9dba209d481040b3bfe6c30de0a169d3/libs/community/langchain_community/document_loaders/sqlalchemy.py#L49-L52